RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/TechnicianService.java

```diff
@@ -277,7 +278,66 @@ public static TechnicianDto technicianToDTO(Technician tech) {
 		
 	}
 	
+	@Transactional // TODO check if this is what is needed for this method: this method supposed to remove absolutely all the timeslots and appointments from technicians shedule
+	public String deleteFullTechnicianWorkSchedule(String email) throws Exception{
+		
+		if(email == null) {
+			throw new Exception("Email cannot be empty.");
+		}
+
+		Technician technician = technicianRepository.findTechnicianByEmail(email);
+		
+		if(technician == null) {
+			throw new Exception("Technician not found.");
+		}
+		
+		// removing all the appointments
+		for (Appointment appointmentToRemove: technician.getAppointments()) {
+			technician.getAppointments().remove(appointmentToRemove);
+		}
+		// removing all the timeslots
+		for (TimeSlot timeSlotToRemove: technician.getTimeslots()) {
+			technician.getTimeslots().remove(timeSlotToRemove);
+		}
+
+		technicianRepository.save(technician);
+		
+		return "All work hour timeslots and associated appointments for technician " + email + " were successfully removed.";
+	}
 	
+	@Transactional // TODO check if this is what is needed for this method: method supposed to remove the specific timeSlot and appointment from technician schedule
+	public String deleteSpecificTechnicianTimeSlotAndAppointmentsFromSchedule(String email, TimeSlot slotToDelete) throws Exception{
+		boolean removedAppointments = false;
+		boolean removedTimeSlot = false;
+		
+		if (email == null) {
+			throw new Exception("Email cannot be empty.");
+		}
+
+		Technician technician = technicianRepository.findTechnicianByEmail(email);
+		if (technician == null) {
+			throw new Exception("Technician not found.");
+		}
+		// First removing all the appointments within the chosen work time slot
+		for (Appointment appointmentToDelete: technician.getAppointments()) {
+			if (appointmentToDelete.getTimeSlot().equals(slotToDelete)) {
+				technician.getAppointments().remove(appointmentToDelete);
+				removedAppointments = true;
```

Comment:
I think we'll need to add a  `break` statement right after finding the `appointment` to delete.

```suggestion
				removedAppointments = true;
				break;
```